executor/join.go

```diff
@@ -313,8 +319,85 @@ func (e *HashJoinExec) handleJoinWorkerPanic(r interface{}) {
 	e.joinWorkerWaitGroup.Done()
 }
 
+// Handling unmatched rows from the hash table
+func (e *HashJoinExec) handleUnmatchedRowsFromHashTableInMemory(workerID uint) {
+	ok, joinResult := e.getNewJoinResult(workerID)
+	if !ok {
+		return
+	}
+	numChks := e.rowContainer.records.NumChunks()
+	for i := int(workerID); i < numChks; i += int(e.concurrency) {
+		chk := e.rowContainer.records.GetChunk(i)
+		for j := 0; j < chk.NumRows(); j++ {
+			if e.outerMatchedStatus[i].UnsafeIsSet(j) == false { // process unmatched outer rows
+				e.joiners[workerID].onMissMatch(false, chk.GetRow(j), joinResult.chk)
+			}
+			if joinResult.chk.IsFull() {
+				e.joinResultCh <- joinResult
+				ok, joinResult = e.getNewJoinResult(workerID)
+				if !ok {
+					return
+				}
+			}
+		}
+	}
+
+	if joinResult == nil {
+		return
+	} else if joinResult.err != nil || (joinResult.chk != nil && joinResult.chk.NumRows() > 0) {
+		e.joinResultCh <- joinResult
+	}
+}
+
+// Sequentially handling unmatched rows from the hash table
+func (e *HashJoinExec) handleUnmatchedRowsFromHashTableInDisk(workerID uint) {
+	ok, joinResult := e.getNewJoinResult(workerID)
+	if !ok {
+		return
+	}
+	numChks := e.rowContainer.recordsInDisk.NumChunks()
+	for i := 0; i < numChks; i++ {
+		numOfRows := e.rowContainer.recordsInDisk.NumRowsOfChunk(i)
+		for j := 0; j < numOfRows; j++ {
+			row, err := e.rowContainer.recordsInDisk.GetRow(chunk.RowPtr{ChkIdx: uint32(i), RowIdx: uint32(j)})
+			if err != nil {
+				return
```

Comment:
We should not ignore err here.

Reply:
ok, insert the error into the joinResult.err before return.

Reply:
Need to send it to a channel, or it will take no effect.

Reply:
ok